### PR TITLE
feat: add MicroplateSet

### PR DESCRIPTION
**Pull request by simbig**
- [x] Added automated tests
- [x] Documented for all relevant versions
- [x] Updated the changelog

**Changes**

- Add `MicroplateSet` possibility to calculate a `Location` within this set by a position

### CODE REVIEW

File: src/MicroplateSet/MicroplateSet.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            throw new \OutOfRangeException("Expected a position between 1-{$positionsCount}, got: {$setPosition}.");
+        }
+
+        $plateIndex = (int) floor(($setPosition - 1) / $this->coordinateSystem->positionsCount());
+        $positionOnSinglePlate = $setPosition - ($plateIndex * $this->coordinateSystem->positionsCount());
+
+        return new Location(
+            Coordinate::fromPosition($positionOnSinglePlate, $direction, $this->coordinateSystem),
+            $this->plateIDs()[$plateIndex]
+        );
```
Comment (simbig) [inline src/MicroplateSet/MicroplateSet.php:59]: @spawnia can you help me to fix this stan error:

```55     Method Mll\Microplate\MicroplateSet\MicroplateSet::locationFromPosition() should return Mll\Microplate\MicroplateSet\Location<TCoordinateSystem of Mll\Microplate\CoordinateSystem> but returns Mll\Microplate\MicroplateSet\Location<Mll\Microplate\CoordinateSystem>.```

File: src/MicroplateSet/Location.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * @template TCoordinateSystem of CoordinateSystem
+ */
+final class Location
+{
+    public string $plateID;
+
+    /**
+     * @var \Mll\Microplate\Coordinate<TCoordinateSystem>
+     */
+    public $coordinate;
```
Comment (spawnia) [inline src/MicroplateSet/Location.php]:
```suggestion
    public Coordinate $coordinate;
```

File: src/MicroplateSet/Location.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    /**
+     * @param \Mll\Microplate\Coordinate<TCoordinateSystem> $coordinate
+     */
+    public function __construct(Coordinate $coordinate, string $plateId)
+    {
+        $this->coordinate = $coordinate;
+        $this->plateID = $plateId;
```
Comment (spawnia) [inline src/MicroplateSet/Location.php]:
```suggestion
    public function __construct(Coordinate $coordinate, string $plateID)
    {
        $this->coordinate = $coordinate;
        $this->plateID = $plateID;
```

File: src/MicroplateSet/MicroplateSet.php (hunk 1 above)
Comment (spawnia) [reply src/MicroplateSet/MicroplateSet.php:59]: Generic inference is too weak to recognize this code is correct, I would just ignore the error here.

File: src/MicroplateSet/MicroplateSetABCD.php [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+/**
+ * @template TCoordinateSystem of CoordinateSystem
+ *
+ * @phpstan-extends MicroplateSet<TCoordinateSystem>
+ */
+final class MicroplateSetABCD extends MicroplateSet
+{
+    /**
+     * {@inheritDoc}
+     */
```
Comment (spawnia) [inline src/MicroplateSet/MicroplateSetABCD.php]:
```suggestion
```

File: tests/Unit/MicroplateSet/MicroplateSetTest.php [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+            [
+                [
+                    'position' => 1,
+                    'coordinateString' => 'A1',
+                    'plateID' => 'A',
+                ], [
+                    'position' => 2,
+                    'coordinateString' => 'B1',
+                    'plateID' => 'A',
+                ], [
```
Comment (spawnia) [inline tests/Unit/MicroplateSet/MicroplateSetTest.php]:
```suggestion
        yield [
            'position' => 1,
            'coordinateString' => 'A1',
            'plateID' => 'A',
        ];
        yield [
            'position' => 2,
            'coordinateString' => 'B1',
            'plateID' => 'A',
        ];
```

And so on. Reduces indentation by 8. Same in the other data provider.

File: tests/Unit/MicroplateSet/MicroplateSetTest.php [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+        $location = $microplateSet->locationFromPosition($position, FlowDirection::COLUMN());
+        self::assertSame($coordinateString, $location->coordinate->toString());
+        self::assertSame($plateID, $location->plateID);
+    }
+
+    /**
+     * @return list<array{position: int, coordinateString: string, plateID: string}>
+     */
+    public static function dataProvider96Well(): array
```
Comment (spawnia) [inline tests/Unit/MicroplateSet/MicroplateSetTest.php]:
```suggestion
     * @return iterable<array{position: int, coordinateString: string, plateID: string}>
     */
    public static function dataProvider96Well(): iterable
```

**Review by spawnia — CHANGES_REQUESTED** [review]

**Review by spawnia — APPROVED** [review]

**Pull request merged**